### PR TITLE
Fine-tune the Add to cart form block

### PR DESCRIPTION
**Pull request by nefeline**
While working on #8225 @gigitux made some changes that updated the structure of the `product-details` and the `gallery` blocks. This PR updates the `add-to-cart-form` block to ensure the same structure is in place by:

- Moving this block to live under the `product-details` directory (Original discussion [here](https://github.com/woocommerce/woocommerce-blocks/pull/8225#issuecomment-1429639562)).
- Updating the class prefix (Original discussion [here](https://github.com/woocommerce/woocommerce-blocks/pull/8225#discussion_r1106985254)).
- Making this block available only on the Single Product Template (based on his recent addition of `registerBlockSingleProductTemplate` made on #8235 )

### Testing

The test instructions here are identical to #8284 : no changes should be observed in the functionality of this block.

#### User Facing Testing

* Make sure you are using a block theme.
* Make sure you have the following product types created in your store: Simple, Variable, Grouped and External. 
* Edit the Single Product template on the Site Editor ( e.g. path > wp-admin/site-editor.php?postType=wp_template&postId=woocommerce%2Fwoocommerce%2F%2Fsingle-product&canvas=edit&sidebar=%2Ftemplates).
* Insert the new **Add to Cart Form** block and save.
* Make sure in the editor you see a placeholder block and it matches the following:

Screenshot 2023-02-20 at 14 51 16

* Access a single product page for a Simple product and ensure the content matches the one displayed on the screenshot:
Screenshot 2023-01-30 at 20 06 48

* Access a single product page for a Variable product and ensure the content matches the one displayed on the screenshot:

Screenshot 2023-01-30 at 20 05 45

* Access a single product page for a Grouped product and ensure the content matches the one displayed on the screenshot:

Screenshot 2023-01-30 at 20 03 51

Screenshot 2023-01-30 at 20 04 53

* Access a single product page for an External product and ensure the content matches the one displayed on the screenshot:

Screenshot 2023-01-30 at 20 07 33

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> N/A

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		registerBlockType( metadata, {
+			icon: {
+				src: (
+					<Icon
+						icon={ button }
+						className="wc-block-editor-components-block-icon"
+					/>
+				),
+			},
+			attributes: {
```
Comment (gigitux) [inline assets/js/atomic/blocks/product-elements/add-to-cart-form/index.tsx]: If we don't add any attributes, we can remove this key.

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.woocommerce.wp-block-add-to-cart-form {
```
Comment (gigitux) [inline assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: I think that we should avoid custom CSS rules, especially if we will implement Global Styles.

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+	font-size: inherit;
+}
+
+.woocommerce.wp-block-add-to-cart-form .product .summary {
```
Comment (gigitux) [inline assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: This selector is not valid anymore. More details: per0F9-fk-p2#comment-202

**Review by gigitux — APPROVED** [review]
Overall LGTM! I have some doubts about the style. Pre-approving it.

**Review by imanish003 — APPROVED** [review]
I have tested the changes & It looks good to me. Well done @nefeline 🚀

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss (hunk 3 above)
Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: ✅ Done!

I completely removed the selectors/styles for this block: the only consequence of this change is that the font size doesn't match the default expected for the add-to-cart form anymore:

Before:
Screenshot 2023-02-22 at 16 25 23

After:
Screenshot 2023-02-22 at 16 26 10

Are we ok with this difference?

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss (hunk 2 above)
Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: See: https://github.com/woocommerce/woocommerce-blocks/pull/8482#discussion_r1114513682

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/index.tsx (hunk 1 above)
Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/index.tsx]: ✅

**Comment by nefeline** [comment]
Thanks for the review, folks! Feedback addressed.

@gigitux I left a question for you on https://github.com/woocommerce/woocommerce-blocks/pull/8482#discussion_r1114513682

File: assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss (hunk 3 above)
Comment (gigitux) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: I thought the WC Core style applies correctly on this block. Is there a reason why doesn't it happen?

I suggest checking why this happens and if the reason is that the CSS selector defined in WC Core is not valid anymore, restore the styles that you defined. Does it make sense?

Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: > I thought the WC Core style applies correctly on this block. Is there a reason why doesn't it happen?

No, it doesn't: the `.summary` and `.entry-summary` selectors are the ones influencing the font size (and that's the reason why they were added in the first place). Based on per0F9-fk-p2#comment-202, it looks like the decision was to not support this selector anymore. I've re-opened that discussion with a few FU questions.

Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/add-to-cart-form/style.scss]: Porting over the discussion from per0F9-fk-p2#comment-361 :

New styles were added here to account for removing the `.summary` selector. Now the block styles are preserved:

Screenshot 2023-02-24 at 13 58 51

**Review by gigitux — APPROVED** [review]
LGTM 🚢

**Review by thealexandrelara — APPROVED** [review]
LGTM! 🚀

**Pull request merged**